packages/xo-server/src/_pRetry.js

```diff
@@ -0,0 +1,36 @@
+import iteratee from 'lodash/iteratee'
+import pDelay from 'promise-toolbox/delay'
+
+function stopRetry(error) {
+  this.error = error
+  // eslint-disable-next-line no-throw-literal
+  throw this
+}
+
+// do not retry on ReferenceError and TypeError which are programmer errors
+const defaultMatcher = error =>
+  !(error instanceof ReferenceError || error instanceof TypeError)
+
+export default async function pRetry(
+  fn,
+  { delay = 1e3, tries = 10, when } = {}
+) {
+  const container = { error: undefined }
+  const stop = stopRetry.bind(container)
+
+  when = when === undefined ? defaultMatcher : iteratee(when)
+
+  while (true) {
+    try {
+      return await fn(stop)
```

Comment:
Should we add a test for the `stop` function?

Reply:
I did not use it here and I don't have time to do it before the release, I suggest we merge it as is, we'll add the test later if necessary.